### PR TITLE
EES-3974 Make 'Add embed block' button only visible to BAU users.

### PR DESCRIPTION
**Pull request by benoutram**
This PR makes the 'Add embed block' button within the admin's content section editor only visible to BAU users. It will be made accessible to all contributors again at a later date.

To restrict visibility of the button it introduces a new front-end and back-end global permission `IsBauUser` which is determined from a new policy of the same name.

### Other changes

* Slight tidy up of  back-end `PermissionsController` - Introduces new record type view model `GlobalPermissionsViewModel`. Constrains all Guid parameter types in api routes. Stops using the controller route `[controller]` and prefixes all routes with `permissions/` instead which isn't susceptible to breaking if the controller class is renamed.
* Removes back-end user extension method `CheckCanRunMigrations` and security policy `SecurityPolicies.CanRunReleaseMigrations` which are replaced by `CheckIsBauUser`.
* Changes back-end `PublicationService.ListPublicationSummaries` to use permission check `CheckIsBauUser` rather than `CheckCanManageAllUsers`.

### UI test report

image

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationServicePermissionTests.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ await PermissionTestUtils.PolicyCheckBuilder<SecurityPolicies>()
         public async Task ListPublicationSummaries()
         {
             await PermissionTestUtils.PolicyCheckBuilder<SecurityPolicies>()
-                .ExpectCheckToFail(SecurityPolicies.CanManageUsersOnSystem)
+                .ExpectCheckToFail(SecurityPolicies.IsBauUser)
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationServicePermissionTests.cs:58]: Would probably check the policy `CanViewAllPublications` instead, which amounts to the same really, but until we replace all BAU-only SecurityPolicies with IsBauUser, might as well use it here?

File: src/explore-education-statistics-admin/src/pages/release/content/components/ReleaseContentAccordionSection.tsx [hunk 2]
```diff
@@ -232,13 +235,17 @@ const ReleaseContentAccordionSection = ({
...
                   </Button>
                 )}
-                {!showEmbedDashboardForm && (
-                  <Button
-                    variant="secondary"
-                    onClick={toggleEmbedDashboardForm.on}
-                  >
-                    Add embed block
-                  </Button>
+                {user?.permissions.isBauUser && (
```
Comment (duncan-at-hiveit) [inline src/explore-education-statistics-admin/src/pages/release/content/components/ReleaseContentAccordionSection.tsx:238]: I guess that all our Robot tests are using BAU to embed blocks?

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationServicePermissionTests.cs (hunk 1 above)
Comment (benoutram) [reply src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationServicePermissionTests.cs]: Ah yep, I didn't realise that one existed, thanks. I've switched it to use `CanViewAllPublications` now, thanks.

File: src/explore-education-statistics-admin/src/pages/release/content/components/ReleaseContentAccordionSection.tsx (hunk 2 above)
Comment (benoutram) [reply src/explore-education-statistics-admin/src/pages/release/content/components/ReleaseContentAccordionSection.tsx:238]: Yeah, the only UI test covering embed blocks currently is `admin_and_public_2\bau\publish_release_and_amend.robot` and it's using the BAU user.

File: src/explore-education-statistics-admin/src/components/PageFooter.tsx [hunk 3]
```diff
@@ -21,7 +21,7 @@ const PageFooter = ({ wide }: Props) => {
           <div className="govuk-footer__meta-item govuk-footer__meta-item--grow">
             <h2 className="govuk-visually-hidden">Support links</h2>
             <ul className="govuk-footer__inline-list">
-              {user?.permissions.canAccessUserAdministrationPages && (
+              {user?.permissions.isBauUser && (
```
[the remark this answers is not in the document]
Comment (benoutram) [reply src/explore-education-statistics-admin/src/components/PageFooter.tsx:24]: Cool, I didn't know that, thanks. I will mention it to them when this change is merged, so they should know by the time it's ready to go out early next week.

**Review by duncan-at-hiveit — APPROVED** [review]

**Pull request merged**